uniswap-v2/src/lib.rs

```diff
@@ -192,16 +201,255 @@ fn map_swap_events(
+        output.set(0, &pool_key, &proto::encode(&event).unwrap());
+    }
+}
+
+#[substreams::handlers::store]
+fn store_withdraws(withdraws: dex_amm::Withdraws, output: store::StoreSetRaw) {
+    log::info!("Stored events {}", withdraws.items.len());
+    for event in withdraws.items {
+        let pool_key = keyer::withdraw_key(&event.tx_hash, event.log_index);
+        output.set(0, &pool_key, &proto::encode(&event).unwrap());
+    }
+}
+
+#[substreams::handlers::store]
+fn store_swaps(swaps: dex_amm::Swaps, output: store::StoreSetRaw) {
+    log::info!("Stored events {}", swaps.items.len());
+    for event in swaps.items {
         let pool_key = keyer::swap_key(&event.tx_hash, event.log_index);
         output.set(0, &pool_key, &proto::encode(&event).unwrap());
     }
 }
 
 #[substreams::handlers::store]
-fn store_swap_count(swap_events: uniswap::SwapEvents, s: store::StoreAddInt64) {
-    s.add(0, "total", swap_events.items.len() as i64);
+fn store_counts(clock: Clock, pair_created_events: uniswap::PairCreatedEvents, mint_events: uniswap::MintEvents, burn_events: uniswap::BurnEvents, swap_events: uniswap::SwapEvents, s: store::StoreAddInt64) {
+    let timestamp_seconds = clock.timestamp.unwrap().seconds;
+    let day_id: String = (timestamp_seconds / 86400).to_string();
+    let hour_id: String = (timestamp_seconds / 3600).to_string();
+
+    for event in mint_events.items {
+        s.add(event.log_ordinal, &keyer::usage_count_key(&"Deposit".to_string(), &"Day".to_string(), &day_id), 1);
+        s.add(event.log_ordinal, &keyer::usage_count_key(&"Deposit".to_string(), &"Hour".to_string(), &hour_id), 1);
+        s.add (event.log_ordinal, &keyer::transaction_count_key(&"Day".to_string(), &day_id), 1);
+        s.add (event.log_ordinal, &keyer::transaction_count_key(&"Hour".to_string(), &hour_id), 1);
+    }
+
+    for event in burn_events.items {
+        s.add(event.log_ordinal, &keyer::usage_count_key(&"Withdraw".to_string(), &"Day".to_string(), &day_id), 1);
+        s.add(event.log_ordinal, &keyer::usage_count_key(&"Withdraw".to_string(), &"Hour".to_string(), &hour_id), 1);
+        s.add (event.log_ordinal, &keyer::transaction_count_key(&"Day".to_string(), &day_id), 1);
+        s.add (event.log_ordinal, &keyer::transaction_count_key(&"Hour".to_string(), &hour_id), 1);
+    }
+
+    for event in swap_events.items {
+        s.add(event.log_ordinal, &keyer::usage_count_key(&"Swap".to_string(), &"Day".to_string(), &day_id), 1);
+        s.add(event.log_ordinal, &keyer::usage_count_key(&"Swap".to_string(), &"Hour".to_string(), &hour_id), 1);
+        s.add (event.log_ordinal, &keyer::transaction_count_key(&"Day".to_string(), &day_id), 1);
+        s.add (event.log_ordinal, &keyer::transaction_count_key(&"Hour".to_string(), &hour_id), 1);
+    }
+}
+
+#[substreams::handlers::store]
+fn map_volumes_from_swaps(swaps: dex_amm::Swaps, s: store::StoreAddBigInt) {
+    for swap in swaps.items {
```

Comment:
is it possible to refactor this into `substreams_helper`?

Reply:
Yes - definitely.